### PR TITLE
Revamped Away Mission: Caves (needs testing)

### PR DESCRIPTION
**Pull request by MMMiracles**
## About The Pull Request
A complete remake of the original Caves away mission, keeping a similar layout theme from the original while enhancing literally everything else about it.

Need a general idea of what's been changed?

- Brand new layout, taking up the entire 255x255 z-level and using pretty much all of it to some effect.
- 4 separate zones, each getting progressively more dangerous as you delve further into the mission.
- Combat encounters and later on, entire zones are randomly picked from a curated list of locations that change every round. Harder to plan out dedicated routes if the caves never seem to stay the same!
- Rich but dense rocks! The minerals down here are aplenty for actual mining, bear in mind these caves are much stronger than your usual mining zone and require more time to break through. As a side effect, most ranged mining equipment doesn't seem all that useful down here past shooting the local fauna.
- An actual finale area! Can you put an end to whatever caused the disappearance of the mining crew and researchers?

## Why It's Good For The Game

If away missions are going to be enabled, they should be set to a higher standard of mapping quality much like our regular stations while having enough challenge to encourage active teamwork between the crew instead of 1 or 2 people running through it for the best equipment before going back station-side. My main goal was creating something with enough variety and challenge that going into this feels like an actual expedition and not just another ruin for powerlooters to speed through. Hopefully this sets a new standard for these missions and shows people what kind of cool stuff you can do that isn't just stations or ruins.

## Changelog

:cl: MMMiracles
add: The Caves away mission has been completely revamped from the ground up!
add: See your local expeditions officer for more details on setting up an exploration crew.
/:cl:

image
image
image

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/bosses/clockmaster/phase_two_attacks.dm [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+			if(!(locate(/obj/effect/clockmaster_aoe_warning) in target_tile) && !(locate(/mob/living/simple_animal/hostile/boss/clockmaster/phase_two) in target_tile))
+				new /obj/effect/clockmaster_aoe_warning(target_tile)
+			addtimer(CALLBACK(src, .proc/BlastEm), 4 SECONDS)
+
+/datum/action/boss/brain_blast/proc/BlastEm()
+	var/mob/living/simple_animal/armored_individual = boss
+	armored_individual.damage_coeff = list(BRUTE = 1, BURN = 1, TOX = 1, CLONE = 1, STAMINA = 0, OXY = 1)
+	boss.mid_ability = FALSE
+	boss.prevent_goto_movement = FALSE
+	playsound(boss, 'sound/misc/fart.ogg', 25)
```
Comment (RigglePrime) [inline code/modules/mob/living/simple_animal/hostile/bosses/clockmaster/phase_two_attacks.dm:36]: Phenomenal

Comment (MMMiracles) [reply code/modules/mob/living/simple_animal/hostile/bosses/clockmaster/phase_two_attacks.dm:36]: listen i added this to amuse myself while testing and forgot to remove it

**Comment by iprice** [comment]
Round ID: 187222 (Terry)
Server Time: 2022-07-24 21:30:23

ADMIN LOG: Away mission _maps/RandomZLevels/caves.dmm loading failed due to errors.

Same on
Round ID: 187232 (Terry)
Server Time: 2022-07-25 01:28:48

**Comment by MMMiracles** [comment]
Do you have a runtime log? It loads fine locally.

**Pull request closed without merging**